### DISCUSSION TITLE
Load Alpine as a script tag and start it later

### DISCUSSION THREAD

**Discussion by pomartel** [alpinejs/alpine#1710] [3. Feature ideas]
Using Alpine as a module, it is possible to load the library and start it later on using `Alpine.start()`. It would be super useful to be able to do the same when Alpine is loaded as a script. That would allow to load the Alpine script before declaring `Alpine.data` components and remove the need to wrap them in an `alpine:init` event listener.

For reference, this is possible with petite-vue! 😉 https://github.com/vuejs/petite-vue#manual-init

**Comment by danddanddand** [comment]
That new vue lib looks super interesting. They are not mounting by default unless you have init attribute on script tag. 

```
let s
if ((s = document.currentScript) && s.hasAttribute('init')) {
  createApp().mount()
}
```
(https://github.com/vuejs/petite-vue/blob/main/src/index.ts)

I could see maybe doing the opposite in Alpine if enough people had the requirement, it would be pretty trivial to add inverse logic. Not sure what the attribute would be called tho? Interesting tho

**Reply by ryangjchandler** [reply]
`lazy` could work if the feature in wanted.

**Reply by danddanddand** [reply]
Lazy makes me think it’ll call Alpine.start() on first usage or automatically. Think I’ve done too much Hibernate/JPA programming where a collection will be wrapped in a proxy on lazy so it won’t query until used. I really can’t think of good name? Hmm.. this is why I like tailwind so much, takes away burden of naming things.

**Comment by pomartel** [comment]
The same issue is described here: https://github.com/alpinejs/alpine/discussions/1705

**Comment by mikestempien** [comment]
Maybe there's some `Alpine.autostart = false` which we could set before Alpine?
